src/main/java/io/github/railroad/config/Config.java

```diff
@@ -13,9 +13,19 @@
 
 import java.util.Optional;
 
+/**
+ * Parses the JSON config file into individual objects.
+ */
 public class Config implements JsonSerializable<JsonObject> {
+    /**
+     * Stores the settings
+     */
```

Comment:
I don't think it makes sense to add javadoc to private fields, that's very uncommon and especially ones like this are completely unnecessary.